src/Engine/Physics/Trigger.cpp

```diff
@@ -0,0 +1,41 @@
+#include "Trigger.hpp"
+
+#include <btBulletDynamicsCommon.h>
+#include "RigidBody.hpp"
+#include "Shape.hpp"
+
+namespace Physics {
+
+    Trigger::Trigger(Shape* shape) : btCollisionWorld::ContactResultCallback() {
+        trigger = new btCollisionObject();
+        trigger->setCollisionShape(shape->GetShape());
+    }
+
+    // Called with each contact for our own processing. This is where we can
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'addSingleResult' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)